Crash with version of feb 10: RuntimeException checkNotNullParameter

Please search to check for an existing issue (including closed issues, for which the fix may not have yet been released) before opening a new issue: https://github.com/k9mail/k-9/issues?q=is%3Aissue

If you are unsure whether or not you have found a bug please post to the support forum instead: https://forum.k9mail.app

**Describe the bug**
Application does not start

**To Reproduce**
Steps to reproduce the behavior:
1. Click on K9 Icon, application crashes

**Expected behavior**
Application starts

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Environment (please complete the following information):**
 - K-9 Mail version: 5.728
 - Android version: 10
 - Device: Redmi Note 9S
 - Account type: all

**Additional context**
Runtime exception Unable to create application com.fsck.k9.app
org.koin.core.error.InstanceCreationException : Could not create instance for Single 'com.kfsck.k9.Preferences'
somewhere in com.fsck.k9.K9.getPreferences

Caused by:
java.lang.NullPointerException
Parameter specified as non-null is null: method kotlin.jvm.internal.Instrincsics.checkNotNullParameter, parameter authenticationType at com.fsck.k9.mail.ServerSettings.<init>




**Logs**

adb logcat | findstr k9


02-11 07:20:11.048  1793  4196 I ActivityTaskManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.fsck.k9/.activity.MessageList bnds=[587,185][755,353] (has extras)} from uid 10086
02-11 07:20:11.133  1793  1930 I ActivityManager: Start proc 18326:com.fsck.k9/u0a336 for activity {com.fsck.k9/com.fsck.k9.activity.MessageList} caller=com.miui.home
02-11 07:20:11.249 18326 18326 E AndroidRuntime: Process: com.fsck.k9, PID: 18326
02-11 07:20:11.249 18326 18326 E AndroidRuntime: java.lang.RuntimeException: Unable to create application com.fsck.k9.App: org.koin.core.error.InstanceCreationException: Could not create instance for [Single:'com.fsck.k9.Preferences']
02-11 07:20:11.249 18326 18326 E AndroidRuntime: Caused by: org.koin.core.error.InstanceCreationException: Could not create instance for [Single:'com.fsck.k9.Preferences']
02-11 07:20:11.249 18326 18326 E AndroidRuntime:        at com.fsck.k9.K9$$special$$inlined$inject$1.invoke(DI.kt:118)
02-11 07:20:11.249 18326 18326 E AndroidRuntime:        at com.fsck.k9.K9.getPreferences(Unknown Source:2)
02-11 07:20:11.249 18326 18326 E AndroidRuntime:        at com.fsck.k9.K9.init(K9.kt:322)
02-11 07:20:11.249 18326 18326 E AndroidRuntime:        at com.fsck.k9.App.onCreate(App.kt:22)
02-11 07:20:11.249 18326 18326 E AndroidRuntime:        at com.fsck.k9.mail.ServerSettings.<init>(Unknown Source:12)
02-11 07:20:11.249 18326 18326 E AndroidRuntime:        at com.fsck.k9.mail.ServerSettings.<init>(ServerSettings.kt:17)
02-11 07:20:11.249 18326 18326 E AndroidRuntime:        at com.fsck.k9.mail.ServerSettings.<init>(Unknown Source:18)
02-11 07:20:11.249 18326 18326 E AndroidRuntime:        at com.fsck.k9.preferences.migrations.migration12.SmtpTransportUriDecoder.decodeSmtpUri(SmtpTransportUriDecoder.java:99)
02-11 07:20:11.249 18326 18326 E AndroidRuntime:        at com.fsck.k9.preferences.migrations.StorageMigrationTo12.convertTransportUri(StorageMigrationTo12.kt:53)
02-11 07:20:11.249 18326 18326 E AndroidRuntime:        at com.fsck.k9.preferences.migrations.StorageMigrationTo12.removeStoreAndTransportUri(StorageMigrationTo12.kt:29)
02-11 07:20:11.249 18326 18326 E AndroidRuntime:        at com.fsck.k9.preferences.migrations.StorageMigrations.upgradeDatabase(StorageMigrations.kt:20)
02-11 07:20:11.249 18326 18326 E AndroidRuntime:        at com.fsck.k9.preferences.K9StoragePersister.openDB(K9StoragePersister.java:46)
02-11 07:20:11.249 18326 18326 E AndroidRuntime:        at com.fsck.k9.preferences.K9StoragePersister.loadValues(K9StoragePersister.java:145)
02-11 07:20:11.249 18326 18326 E AndroidRuntime:        at com.fsck.k9.Preferences.<init>(Preferences.java:79)
02-11 07:20:11.249 18326 18326 E AndroidRuntime:        at com.fsck.k9.Preferences.getPreferences(Preferences.java:45)
02-11 07:20:11.249 18326 18326 E AndroidRuntime:        at com.fsck.k9.KoinModuleKt$mainModule$1$1.invoke(KoinModule.kt:16)
02-11 07:20:11.249 18326 18326 E AndroidRuntime:        at com.fsck.k9.KoinModuleKt$mainModule$1$1.invoke(Unknown Source:4)
02-11 07:20:11.254  1793  4196 W ActivityTaskManager:   Force finishing activity com.fsck.k9/.activity.MessageList
02-11 07:20:11.300  1793  1839 I ActivityManager: Process com.fsck.k9 (pid 18326) has died: vis+99 TOP
02-11 07:20:11.755  1793  1907 W ActivityTaskManager: Activity top resumed state loss timeout for ActivityRecord{34050e1 u0 com.fsck.k9/.activity.MessageList t-1 f}
02-11 07:32:07.563  1793  3174 I ActivityTaskManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.fsck.k9/.activity.MessageList bnds=[586,184][754,352] (has extras)} from uid 10086
02-11 07:32:07.649  1793  1930 I ActivityManager: Start proc 20726:com.fsck.k9/u0a336 for activity {com.fsck.k9/com.fsck.k9.activity.MessageList} caller=com.miui.home
02-11 07:32:07.766 20726 20726 E AndroidRuntime: Process: com.fsck.k9, PID: 20726
02-11 07:32:07.766 20726 20726 E AndroidRuntime: java.lang.RuntimeException: Unable to create application com.fsck.k9.App: org.koin.core.error.InstanceCreationException: Could not create instance for [Single:'com.fsck.k9.Preferences']
02-11 07:32:07.766 20726 20726 E AndroidRuntime: Caused by: org.koin.core.error.InstanceCreationException: Could not create instance for [Single:'com.fsck.k9.Preferences']
02-11 07:32:07.766 20726 20726 E AndroidRuntime:        at com.fsck.k9.K9$$special$$inlined$inject$1.invoke(DI.kt:118)
02-11 07:32:07.766 20726 20726 E AndroidRuntime:        at com.fsck.k9.K9.getPreferences(Unknown Source:2)
02-11 07:32:07.766 20726 20726 E AndroidRuntime:        at com.fsck.k9.K9.init(K9.kt:322)
02-11 07:32:07.766 20726 20726 E AndroidRuntime:        at com.fsck.k9.App.onCreate(App.kt:22)
02-11 07:32:07.766 20726 20726 E AndroidRuntime:        at com.fsck.k9.mail.ServerSettings.<init>(Unknown Source:12)
02-11 07:32:07.766 20726 20726 E AndroidRuntime:        at com.fsck.k9.mail.ServerSettings.<init>(ServerSettings.kt:17)
02-11 07:32:07.766 20726 20726 E AndroidRuntime:        at com.fsck.k9.mail.ServerSettings.<init>(Unknown Source:18)
02-11 07:32:07.766 20726 20726 E AndroidRuntime:        at com.fsck.k9.preferences.migrations.migration12.SmtpTransportUriDecoder.decodeSmtpUri(SmtpTransportUriDecoder.java:99)
02-11 07:32:07.766 20726 20726 E AndroidRuntime:        at com.fsck.k9.preferences.migrations.StorageMigrationTo12.convertTransportUri(StorageMigrationTo12.kt:53)
02-11 07:32:07.766 20726 20726 E AndroidRuntime:        at com.fsck.k9.preferences.migrations.StorageMigrationTo12.removeStoreAndTransportUri(StorageMigrationTo12.kt:29)
02-11 07:32:07.766 20726 20726 E AndroidRuntime:        at com.fsck.k9.preferences.migrations.StorageMigrations.upgradeDatabase(StorageMigrations.kt:20)
02-11 07:32:07.766 20726 20726 E AndroidRuntime:        at com.fsck.k9.preferences.K9StoragePersister.openDB(K9StoragePersister.java:46)
02-11 07:32:07.766 20726 20726 E AndroidRuntime:        at com.fsck.k9.preferences.K9StoragePersister.loadValues(K9StoragePersister.java:145)
02-11 07:32:07.766 20726 20726 E AndroidRuntime:        at com.fsck.k9.Preferences.<init>(Preferences.java:79)
02-11 07:32:07.766 20726 20726 E AndroidRuntime:        at com.fsck.k9.Preferences.getPreferences(Preferences.java:45)
02-11 07:32:07.766 20726 20726 E AndroidRuntime:        at com.fsck.k9.KoinModuleKt$mainModule$1$1.invoke(KoinModule.kt:16)
02-11 07:32:07.766 20726 20726 E AndroidRuntime:        at com.fsck.k9.KoinModuleKt$mainModule$1$1.invoke(Unknown Source:4)
02-11 07:32:07.770  1793 15197 W ActivityTaskManager:   Force finishing activity com.fsck.k9/.activity.MessageList
02-11 07:32:07.811  1793  4398 I ActivityManager: Process com.fsck.k9 (pid 20726) has died: vis+99 TOP
02-11 07:32:08.272  1793  1907 W ActivityTaskManager: Activity top resumed state loss timeout for ActivityRecord{7afa451 u0 com.fsck.k9/.activity.MessageList t-1 f}
02-11 07:36:43.199  1793  2365 I ActivityTaskManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.fsck.k9/.activity.MessageList bnds=[588,187][756,355] (has extras)} from uid 10086
02-11 07:36:43.255  1793  1930 I ActivityManager: Start proc 21800:com.fsck.k9/u0a336 for activity {com.fsck.k9/com.fsck.k9.activity.MessageList} caller=com.miui.home
02-11 07:36:43.387 21800 21800 E AndroidRuntime: Process: com.fsck.k9, PID: 21800
02-11 07:36:43.387 21800 21800 E AndroidRuntime: java.lang.RuntimeException: Unable to create application com.fsck.k9.App: org.koin.core.error.InstanceCreationException: Could not create instance for [Single:'com.fsck.k9.Preferences']
02-11 07:36:43.387 21800 21800 E AndroidRuntime: Caused by: org.koin.core.error.InstanceCreationException: Could not create instance for [Single:'com.fsck.k9.Preferences']
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.K9$$special$$inlined$inject$1.invoke(DI.kt:118)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.K9.getPreferences(Unknown Source:2)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.K9.init(K9.kt:322)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.App.onCreate(App.kt:22)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.mail.ServerSettings.<init>(Unknown Source:12)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.mail.ServerSettings.<init>(ServerSettings.kt:17)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.mail.ServerSettings.<init>(Unknown Source:18)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.preferences.migrations.migration12.SmtpTransportUriDecoder.decodeSmtpUri(SmtpTransportUriDecoder.java:99)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.preferences.migrations.StorageMigrationTo12.convertTransportUri(StorageMigrationTo12.kt:53)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.preferences.migrations.StorageMigrationTo12.removeStoreAndTransportUri(StorageMigrationTo12.kt:29)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.preferences.migrations.StorageMigrations.upgradeDatabase(StorageMigrations.kt:20)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.preferences.K9StoragePersister.openDB(K9StoragePersister.java:46)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.preferences.K9StoragePersister.loadValues(K9StoragePersister.java:145)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.Preferences.<init>(Preferences.java:79)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.Preferences.getPreferences(Preferences.java:45)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.KoinModuleKt$mainModule$1$1.invoke(KoinModule.kt:16)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.KoinModuleKt$mainModule$1$1.invoke(Unknown Source:4)
02-11 07:36:43.392  1793  4196 W ActivityTaskManager:   Force finishing activity com.fsck.k9/.activity.MessageList
02-11 07:36:43.431  1793  2365 I ActivityManager: Process com.fsck.k9 (pid 21800) has died: vis+99 TOP
02-11 07:36:43.894  1793  1907 W ActivityTaskManager: Activity top resumed state loss timeout for ActivityRecord{5cf0f97 u0 com.fsck.k9/.activity.MessageList t-1 f}



adb logcat | findstr 21800


02-11 07:36:43.255  1793  1930 I ActivityManager: Start proc 21800:com.fsck.k9/u0a336 for activity {com.fsck.k9/com.fsck.k9.activity.MessageList} caller=com.miui.home
02-11 07:36:43.387 21800 21800 E AndroidRuntime: FATAL EXCEPTION: main
02-11 07:36:43.387 21800 21800 E AndroidRuntime: Process: com.fsck.k9, PID: 21800
02-11 07:36:43.387 21800 21800 E AndroidRuntime: java.lang.RuntimeException: Unable to create application com.fsck.k9.App: org.koin.core.error.InstanceCreationException: Could not create instance for [Single:'com.fsck.k9.Preferences']
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6587)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at android.app.ActivityThread.access$1400(ActivityThread.java:224)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1887)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:107)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:224)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:7562)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
02-11 07:36:43.387 21800 21800 E AndroidRuntime: Caused by: org.koin.core.error.InstanceCreationException: Could not create instance for [Single:'com.fsck.k9.Preferences']
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at org.koin.core.instance.InstanceFactory.create(InstanceFactory.kt:59)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at org.koin.core.instance.SingleInstanceFactory.create(SingleInstanceFactory.kt:40)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at org.koin.core.instance.SingleInstanceFactory.get(SingleInstanceFactory.kt:48)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at org.koin.core.registry.InstanceRegistry.resolveInstance$koin_core(InstanceRegistry.kt:87)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at org.koin.core.scope.Scope.resolveInstance(Scope.kt:214)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at org.koin.core.scope.Scope.get(Scope.kt:181)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.K9$$special$$inlined$inject$1.invoke(DI.kt:118)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.K9.getPreferences(Unknown Source:2)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.K9.init(K9.kt:322)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.App.onCreate(App.kt:22)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1190)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6582)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        ... 8 more
**02-11 07:36:43.387 21800 21800 E AndroidRuntime: Caused by: java.lang.NullPointerException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkNotNullParameter, parameter authenticationType**
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.mail.ServerSettings.<init>(Unknown Source:12)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.mail.ServerSettings.<init>(ServerSettings.kt:17)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.mail.ServerSettings.<init>(Unknown Source:18)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.preferences.migrations.migration12.SmtpTransportUriDecoder.decodeSmtpUri(SmtpTransportUriDecoder.java:99)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.preferences.migrations.StorageMigrationTo12.convertTransportUri(StorageMigrationTo12.kt:53)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.preferences.migrations.StorageMigrationTo12.removeStoreAndTransportUri(StorageMigrationTo12.kt:29)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.preferences.migrations.StorageMigrations.upgradeDatabase(StorageMigrations.kt:20)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.preferences.K9StoragePersister.openDB(K9StoragePersister.java:46)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.preferences.K9StoragePersister.loadValues(K9StoragePersister.java:145)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.Preferences.<init>(Preferences.java:79)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.Preferences.getPreferences(Preferences.java:45)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.KoinModuleKt$mainModule$1$1.invoke(KoinModule.kt:16)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at com.fsck.k9.KoinModuleKt$mainModule$1$1.invoke(Unknown Source:4)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        at org.koin.core.instance.InstanceFactory.create(InstanceFactory.kt:50)
02-11 07:36:43.387 21800 21800 E AndroidRuntime:        ... 20 more
02-11 07:36:43.431  1793  2365 I ActivityManager: Process com.fsck.k9 (pid 21800) has died: vis+99 TOP

--- a/app/storage/src/main/java/com/fsck/k9/preferences/migrations/migration12/ImapStoreUriDecoder.java
+++ b/app/storage/src/main/java/com/fsck/k9/preferences/migrations/migration12/ImapStoreUriDecoder.java
@@ -36,8 +36,8 @@ public class ImapStoreUriDecoder {
         String host;
         int port;
         ConnectionSecurity connectionSecurity;
-        AuthType authenticationType = null;
-        String username = null;
+        AuthType authenticationType = AuthType.PLAIN;
+        String username = "";
         String password = null;
         String clientCertificateAlias = null;
         String pathPrefix = null;
@@ -94,13 +94,11 @@ public class ImapStoreUriDecoder {
                     authenticationType = AuthType.valueOf(userInfoParts[0]);
                     username = decodeUtf8(userInfoParts[1]);
                 } else {
-                    authenticationType = AuthType.PLAIN;
                     username = decodeUtf8(userInfoParts[0]);
                 }
             } else if (userInfoParts.length == 2) {
                 // Old/standard style of encoding - PLAIN auth only:
                 // username:password
-                authenticationType = AuthType.PLAIN;
                 username = decodeUtf8(userInfoParts[0]);
                 password = decodeUtf8(userInfoParts[1]);
             } else if (userInfoParts.length == 3) {

--- a/app/storage/src/main/java/com/fsck/k9/preferences/migrations/migration12/Pop3StoreUriDecoder.java
+++ b/app/storage/src/main/java/com/fsck/k9/preferences/migrations/migration12/Pop3StoreUriDecoder.java
@@ -35,7 +35,7 @@ public class Pop3StoreUriDecoder {
         String host;
         int port;
         ConnectionSecurity connectionSecurity;
-        String username = null;
+        String username = "";
         String password = null;
         String clientCertificateAlias = null;
 

--- a/app/storage/src/main/java/com/fsck/k9/preferences/migrations/migration12/SmtpTransportUriDecoder.java
+++ b/app/storage/src/main/java/com/fsck/k9/preferences/migrations/migration12/SmtpTransportUriDecoder.java
@@ -31,8 +31,8 @@ public class SmtpTransportUriDecoder {
         String host;
         int port;
         ConnectionSecurity connectionSecurity;
-        AuthType authType = null;
-        String username = null;
+        AuthType authType = AuthType.PLAIN;
+        String username = "";
         String password = null;
         String clientCertificateAlias = null;
 
@@ -78,10 +78,8 @@ public class SmtpTransportUriDecoder {
         if (smtpUri.getUserInfo() != null) {
             String[] userInfoParts = smtpUri.getUserInfo().split(":");
             if (userInfoParts.length == 1) {
-                authType = AuthType.PLAIN;
                 username = decodeUtf8(userInfoParts[0]);
             } else if (userInfoParts.length == 2) {
-                authType = AuthType.PLAIN;
                 username = decodeUtf8(userInfoParts[0]);
                 password = decodeUtf8(userInfoParts[1]);
             } else if (userInfoParts.length == 3) {

--- a/app/storage/src/main/java/com/fsck/k9/preferences/migrations/migration12/WebDavStoreUriDecoder.java
+++ b/app/storage/src/main/java/com/fsck/k9/preferences/migrations/migration12/WebDavStoreUriDecoder.java
@@ -1,6 +1,7 @@
 package com.fsck.k9.preferences.migrations.migration12;
 
 
+import com.fsck.k9.mail.AuthType;
 import com.fsck.k9.mail.ConnectionSecurity;
 import com.fsck.k9.mail.ServerSettings;
 
@@ -114,6 +115,6 @@ public class WebDavStoreUriDecoder {
         extra.put("authPath", authPath);
         extra.put("mailboxPath", mailboxPath);
 
-        return new ServerSettings("webdav", host, port, connectionSecurity, null, username, password, null, extra);
+        return new ServerSettings("webdav", host, port, connectionSecurity, AuthType.PLAIN, username, password, null, extra);
     }
 }

